worlds/psychonauts/Rules.py

```diff
@@ -0,0 +1,506 @@
+
+            RegionName.WWMAV2: lambda state: self.has_priceless_coin(state) and self.has_telekinesis(state),
+
+            RegionName.WWMAV3: self.has_musket,
+
+            RegionName.WWMADone: self.has_levitation,
+
+            RegionName.BVRB: self.has_edgar_mind,
+
+            RegionName.BVRBLev: self.has_levitation,
+
+            RegionName.BVRBTele: self.has_telekinesis,
+
+            RegionName.BVRBDuster: self.has_cobweb_duster,
+
+            RegionName.BVRBLogs: self.has_pyrokinesis,
+
+            RegionName.BVESLev: self.has_levitation,
+
+            RegionName.BVESCobra: self.has_confusion,
+
+            RegionName.BVESBoss: self.has_telekinesis,
+
+            RegionName.MCTC: lambda state: self.has_cobweb_duster(state) and self.has_oly_mind(state),
+
+            RegionName.MCTCLev: self.has_levitation,
+
+            RegionName.MCTCEscort: self.has_telekinesis,
+
+        }
+
+    def has_button(self, state: CollectionState) -> bool:
+        return state.has(ItemName.SashaButton, self.player)
+
+    def has_lili_bracelet(self, state: CollectionState) -> bool:
+        return state.has(ItemName.LilisBracelet, self.player)
+
+    def has_squirrel_dinner(self, state: CollectionState) -> bool:
+        return state.has(ItemName.SquirrelDinner, self.player)
+
+    def has_oarsmans_badge(self, state: CollectionState) -> bool:
+        return state.has(ItemName.OarsmansBadge, self.player)
+
+    def has_lungfish_call(self, state: CollectionState) -> bool:
+        return state.has(ItemName.LungfishCall, self.player)
+
+    def has_cake(self, state: CollectionState) -> bool:
+        return state.has(ItemName.Cake, self.player)
+
+    def has_coach_mind(self, state: CollectionState) -> bool:
+        return state.has(ItemName.CoachMind, self.player)
+
+    def has_sasha_mind(self, state: CollectionState) -> bool:
+        return state.has(ItemName.SashaMind, self.player)
+
+    def has_milla_mind(self, state: CollectionState) -> bool:
+        return state.has(ItemName.MillaMind, self.player)
+
+    def has_linda_mind(self, state: CollectionState) -> bool:
+        return state.has(ItemName.LindaMind, self.player)
+
+    def has_boyd_mind(self, state: CollectionState) -> bool:
+        return state.has(ItemName.BoydMind, self.player)
+
+    def has_gloria_mind(self, state: CollectionState) -> bool:
+        return state.has(ItemName.GloriaMind, self.player)
+
+    def has_fred_mind(self, state: CollectionState) -> bool:
+        return state.has(ItemName.FredMind, self.player)
+
+    def has_edgar_mind(self, state: CollectionState) -> bool:
+        return state.has(ItemName.EdgarMind, self.player)
+
+    def has_oly_mind(self, state: CollectionState) -> bool:
+        return state.has(ItemName.OlyMind, self.player)
+
+    def has_most_sasha_mind_access(self, state: CollectionState) -> bool:
+        return state.has_all([ItemName.SashaMind, ItemName.Marksmanship], self.player)
```

Comment:
For additional reference, this is a topic that came up recently on the Archipelago discord since using tuples for immutable sequences is known to be faster and more memory efficient. I personally doubt that there will be any noticeable performance difference, but it's such a simple change that it may as well be done.

Reply:
And you get to be the first world to literally ever use tuples for these!